### DISCUSSION TITLE
how to Manage correctly panels position and size using Grafonnet ?

### DISCUSSION THREAD

**Discussion by xinity** [grafana/grafonnet#173] [Q&A]
hello grafonnet community :)

I'm dinging more and more into grafonnet and founding it very useful to do dashboard as code rather than clickops :)

One thing i'm struggling is how to manage correctly my panels size and position in a grid as not all of those panels will have the same size, some will be larger than the others.

when i use the makegrid function with panel options, all the panels have the same size even though i'm trying to fix the gridpos feature for each of them.

any hints to make it correctly will be greatly appreciated TBH :)

**Comment by xinity** [comment]
adding more information, 
What I want to achieve is to have the ability to manage each panel size separately defined in a row which is itself defined in a grid.

**Reply by Duologic** [reply]
`util.grid.wrapPanels` implements a different strategy, other functions in util could help build your own strategy: https://grafana.github.io/grafonnet/API/util.html

https://github.com/grafana/grafonnet/blob/main/custom/util/grid.libsonnet#L30 shows how these can be used.

**Comment by xinity** [comment]
just discovered i failed in the same way as this issue : https://github.com/grafana/grafonnet/issues/145

@Duologic i'm sure i've understood how to leverage: https://grafana.github.io/grafonnet/API/util.html#fn-panelresolvecollapsedflagonrows

any hints perhaps :(

**Reply by Duologic** [reply]
Please show some code.

**Reply by xinity** [reply]
@Duologic i've found how to uses it finally :)

had a little bug but someone on slack fixed it for me.
